tensorboard/db.py

```diff
@@ -38,14 +45,30 @@ class TensorBase(object):
   This class is thread safe.
   """
 
-  def __init__(self, db_connection_provider):
+  def __init__(self, db_connection_provider, retrier_factory=util.Retrier):
     """Creates new instance.
 
     :type db_connection_provider: () -> sqlite3.Connection
+    :type retrier_factory: ((Exception) -> bool) -> util.Retrier
     """
     self._db_connection_provider = db_connection_provider
     self._plugin_ids_by_name = {}  # type: dict[str, int]
     self._plugin_ids_by_name_lock = threading.Lock()
+    self._retrier = retrier_factory(_is_transient_sqlite_error)
+
+  def run_transaction(self, callback):
```

Comment:
It can be tuned in a cumbersome way by providing a different `retrier_factory` to the constructor. I like your thinking here, but I'm not sure if it justifies the API complexity. If you can think about any parts of our data model that could have contention like that, I would love to hear about it. Part of our duty is to anticipate such problems, so we're never faced with that type of decision.

I could only envision a write transaction legitimately failing longer than 30 seconds if the admin is running an `ALTER TABLE` query. I spoke with Martin about this some time back. We both agree that it's OK for this service to have planned downtime. We're aiming for it to be reliable, but we're not aiming for high availability. In other words, it will probably have 99% uptime, but not 99.999%. We would rather have the service be as accessible as possible, to as many users as possible, and grant them a lot of flexibility in terms of choosing their storage layer.